### PR TITLE
Python 37+ initial migration

### PR DESCRIPTION
**Pull request by penguinolog**
##### Checklist
- [X] I've ensured that similar functionality has not already been implemented
- [X] I've ensured that similar functionality has not earlier been proposed and declined
- [X] I've branched off the `master` or `python-dual-support` branch
- [X] I've merged fresh upstream into my branch recently
- [X] I've ran `tox` successfully in local environment

Related #394

### CODE REVIEW

**Comment by penguinolog** [comment]
rebased over test fixes PR

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,17 @@
 language: python
-dist: bionic
+#dist: bionic
```
Comment (penguinolog) [inline .travis.yml:2]: use actual default (BTW better to sunset Travis)

File: docs/manual/wcur2.py
Comment (penguinolog) [inline docs/manual/wcur2.py:1]: Looks like never completed file, syntax is incorrect

File: docs/manual/wcur2.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-        assert row == 0
-        self.cursor_x = col
-        return True
+# def get_pref_col(self, (maxcol,)):
+#         return self.cursor_x
+#
+# def move_cursor_to_coords(self, (maxcol,), col, row):
+#     assert row == 0
+#     self.cursor_x = col
+#     return True
```
Comment (wardi) [inline docs/manual/wcur2.py:7]: commenting out manual examples?

File: examples/asyncio_socket_server.py [hunk 3]
```diff
@@ -4,22 +4,23 @@
 This code works on older Python 3.x if you install `asyncio` from PyPI, and
 even Python 2 if you install `trollius`!
 """
-from __future__ import print_function
 
 try:
     import asyncio
 except ImportError:
-    import trollius as asyncio
+from __future__ import annotations
```
Comment (wardi) [inline examples/asyncio_socket_server.py:11]: indentation error?

File: examples/asyncio_socket_server.py [hunk 4]
```diff
@@ -4,22 +4,23 @@
...
 
-from datetime import datetime
+import logging
 import sys
 import weakref
+from datetime import datetime
+
+import trollius as asyncio
```
Comment (wardi) [inline examples/asyncio_socket_server.py:18]: trollius is only for python2?

File: urwid/web_display.py [hunk 5]
```diff
@@ -51,7 +53,7 @@
 //    License along with this library; if not, write to the Free Software
 //    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 //
-// Urwid web site: http://excess.org/urwid/
+// Urwid web site: https://urwid.org/
```
Comment (wardi) [inline urwid/web_display.py:56]: only update the link in this file?

**Comment by wardi** [comment]
Added my comments. Some good changes here but mostly code reformatting.

File: docs/manual/wcur2.py (hunk 2 above)
Comment (penguinolog) [reply docs/manual/wcur2.py:7]: Problem, that it's part of class copy-paste without class declaration -> this code causes parsing error

File: urwid/curses_display.py [hunk 6]
```diff
@@ -368,7 +372,7 @@ def _encode_mouse_event(self):
         l = []
         def append_button( b ):
             b |= mod
-            l.extend([ 27, ord2('['), ord2('M'), b+32, x+33, y+33 ])
+            l.extend([ 27, '[', 'M', b+32, x+33, y+33 ])
```
[the remark this answers is not in the document]
Comment (penguinolog) [reply urwid/curses_display.py:375]: `ord2` for python3 was `lambda x: x`: https://github.com/urwid/urwid/blob/master/urwid/compat.py#L36
so type was not changed

File: urwid/compat.py [hunk 7]
```diff
@@ -20,88 +19,24 @@
...
-import sys
-
-
-PYTHON3 = sys.version_info > (3, 0)
 
 # for iterating over byte strings:
 # ord2 calls ord in python2 only
 # chr2 converts an ordinal value to a length-1 byte string
 # B returns a byte string in all supported python versions
 # bytes3 creates a byte string from a list of ordinal values
```
Comment (wardi) [inline urwid/compat.py:27]: remove old comments too?

File: urwid/web_display.py (hunk 5 above)
Comment (penguinolog) [reply urwid/web_display.py:56]: fixed all

File: examples/asyncio_socket_server.py (hunk 3 above)
Comment (penguinolog) [reply examples/asyncio_socket_server.py:8]: yes, during auto-format.
Fixed.

File: urwid/compat.py (hunk 7 above)
Comment (penguinolog) [reply urwid/compat.py]: fixed

File: urwid/display_common.py [hunk 8]
```diff
@@ -611,7 +611,7 @@ def _colors(self):
             return 256
         if self._value & (_BG_TRUE_COLOR | _FG_TRUE_COLOR):
             return 2**24
-        if self._value & (_BG_BASIC_COLOR | _BG_BASIC_COLOR):
```
Comment (penguinolog) [inline urwid/display_common.py:572]: trivial typo

File: docs/manual/wcur2.py [hunk 9]
```diff
@@ -1,7 +1,13 @@
...
+
+
+class Widget(urwid.ListBox):
+    def get_pref_col(self, size):
+        return self.cursor_x
+
+    def move_cursor_to_coords(self, size, col, row):
+        assert row == 0
+        self.cursor_x = col
+        return True
```
Comment (wardi) [inline docs/manual/wcur2.py:13]: looks like in docs/manual/widgets.rst where wdur2.py is referenced we can add a `:lines: 7-` to skip the first 6 lines of this file in the docs https://devopstutodoc.readthedocs.io/en/stable/documentation/doc_generators/sphinx/rest_sphinx/code/literalinclude/literalinclude.html#lines-xxx

**Comment by wardi** [comment]
Since some of these changes affect the docs, try regenerating the docs to see if any unintended changes appear

File: docs/manual/wcur2.py (hunk 9 above)
Comment (penguinolog) [reply docs/manual/wcur2.py:13]: Done. Validated build.
Had to merge another PR (#424) due to broken documentation build in master (Issue #509)

**Comment by wardi** [comment]
@penguinolog I don't have any other comments

**Pull request merged**